projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/hip/stream.cpp

```diff
@@ -75,14 +75,28 @@ get_stream_map()
     return _v;
 }
 
+auto&
+get_idx_offset()
+{
+    static auto*& _v = common::static_object<uint64_t>::construct(0ul);
+    return *_v;
+}
+
+auto&
+get_stream_thread_counter()
+{
+    static auto*& _v = common::static_object<std::atomic<uint64_t>>::construct(0ul);
+    return *_v;
+}
+
 auto
 add_stream(hipStream_t stream)
 {
     return get_stream_map()->wlock(
         [](stream_map_t& _data, hipStream_t _stream) {
-            static uint64_t idx_offset = 0;
+            auto& idx_offset = get_idx_offset();
 
-            auto idx = _data.size() + idx_offset;
+            auto idx = _data.size() + idx_offset + get_stream_thread_counter().load();
```

Comment:
I genuinely don't know how this compiles... you are adding an integer to a reference to `static_object<uint64_t>`

Reply:
Nevermind was thinking this was `Synchronized<T>`